lib/main.js

```diff
@@ -121,6 +121,12 @@ function initSession(shiny, options, shinyServer) {
 
       factory(url, ctx, (err, conn) => {
         pc.resolve(err, conn);
+        conn.onclose = e => {
+          if (!options.reconnect) {
+            log("connection closed; reconnect not enabled");
+            reconnectUI.showDisconnected();
```

Comment:
Yes, these two reconnect mechanisms are independent. 

The reconnect that we're talking about (`options.reconnect`) is for hiding hiccups in the underlying SockJS connection from the wrapper in both the client and server. This restores connection to your existing session in a Shiny process.

There is another mode of reconnects that Shiny supports (`pc.recconect`). In this mode, you're starting a new session, rather than reconnecting to your existing one. This is used for stateless apps like dashboards where -- if you hopped on a plane and were offline for 6 hours -- you should just be able to open up the app and have it be alive without concern of whether or not it's a new session.

So I think this behavior is OK.